corehq/apps/ota/views.py

```diff
@@ -302,19 +302,26 @@ def heartbeat(request, domain, app_build_id):
     build_profile_id = request.GET.get('build_profile_id', '')
 
     info = {"app_id": app_id}
+    config = GlobalAppConfig.by_app_id(domain, app_id)
+
+    def _get_version_info(app_config):
+        return {
+            "latest_apk_version": config.get_latest_apk_version(),
+            "latest_ccz_version": config.get_latest_app_version(build_profile_id),
+        }
+
     try:
+        version_info = _get_version_info(config)
         # mobile will send master app_id
-        config = GlobalAppConfig.by_app_id(domain, app_id)
     except (Http404, AssertionError):
         # If it's not a valid master app id, find it by talking to couch
         notify_exception(request, 'Received an invalid heartbeat request')
         app = get_app_cached(domain, app_build_id)
         config = GlobalAppConfig.by_app_id(domain, app.master_id)
+        version_info = _get_version_info(config)
```

Comment:
why the change in behaviour to send this each time? Looks like the mobile currently does not expect it in case it's not master app, so why send it? Just thinking about if it would change things on mobile.

Reply:
yea mean why remove the 'else'? Looking at the code the 'else' block was serving no function at all so I think it wasn't meant to be there.

Reply:
If i am reading correctly, it was meant to run if there was no exception. The config variable was unused so that can be removed in the old implementation.

Reply:
If you look at the original implementation it does update the request: https://github.com/dimagi/commcare-hq/blame/adc47bb934d4e9b75fef669d1ab55392adbceeed/corehq/apps/ota/views.py#L311

Reply:
I see, looks like it fetched it but it never put it in the `info` that was returned?

Reply:
sorry, here's a better link: https://github.com/dimagi/commcare-hq/blob/8a323bef0069ef2ed944018485d8324e141d3437/corehq/apps/ota/views.py#L274

Reply:
oh cool.